lib/merchant.rb

```diff
@@ -32,6 +32,19 @@ def revenue(date=nil)
     end
   end
 
+  def favorite_customer
+    @c = successful_customers.reduce(Hash.new(0)) do |hash, customer_id|
+      hash[customer_id] += 1
+      hash
+    end
+    customer_id, invoice_count = @c.max_by do |key, value|
```

Comment:
Unused block argument - `key`. If it's necessary, use `_` or `_key` as an argument name to indicate that it won't be used.<br>Useless assignment to variable - `invoice_count`. Use `_` or `_invoice_count` as a variable name to indicate that it won't be used.